src/Nethermind/Nethermind.DataMarketplace.Subprotocols/NdmSubprotocol.cs

```diff
@@ -210,14 +210,8 @@ public override void HandleMessage(Packet message)
             }
             
             Logger.Warn($"GETTING MESSAGE: ndm.{NdmMessageCode.GetDescription(message.PacketType)}");
-            try
-            {
-                MessageHandlers[message.PacketType](message);
-            }
-            catch(KeyNotFoundException)
-            {
-                Logger.Error("$There is no handler registered for this message");
-            }
+
+            MessageHandlers[message.PacketType](message);
```

Comment:
You want exception to propagate here?

I was thinking of:

if (MessageHandlers.TryGetValue(message.PacketType, out var handler)
{
  handler(message);
}
else
{
Logger.Error("$There is no handler registered for this message");
}